modules/ml/src/main/java/org/apache/ignite/ml/nn/trainers/local/MLPLocalBatchTrainer.java

```diff
@@ -0,0 +1,210 @@
+            Matrix input = batch.get1();
+            Matrix truth = batch.get2();
+
+            updaterParams = updater.updateParams(mlp, updaterParams, i, input, truth);
+
+            // Update mlp with updater parameters.
+            updaterParams.updateMLP(mlp);
+
+            Matrix predicted = mlp.apply(input);
+
+            int batchSize = input.columnSize();
+
+            err = MatrixUtil.zipFoldByColumns(predicted, truth, (predCol, truthCol) -> loss.apply(truthCol).apply(predCol)).sum() / batchSize;
+
+            debug("Error: " + err);
+
+            if (err < errorThreshold)
+                break;
+
+            i++;
+        }
+
+        return mlp;
+    }
+
+    /**
+     * Construct new trainer with the same parameters as this trainer, but with new loss.
+     *
+     * @param loss New loss function.
+     * @return new trainer with the same parameters as this trainer, but with new loss.
+     */
+    public MLPLocalBatchTrainer withLoss(IgniteFunction<Vector, IgniteDifferentiableVectorToDoubleFunction> loss) {
+        return new MLPLocalBatchTrainer<>(loss, updaterSupplier, errorThreshold, maxIterations);
+    }
+
+    /**
+     * Construct new trainer with the same parameters as this trainer, but with new updater supplier.
+     *
+     * @param updaterSupplier New updater supplier.
+     * @return new trainer with the same parameters as this trainer, but with new updater supplier.
+     */
+    public MLPLocalBatchTrainer withUpdater(IgniteSupplier<MLPParameterUpdater> updaterSupplier) {
+        return new MLPLocalBatchTrainer(loss, updaterSupplier, errorThreshold, maxIterations);
+    }
+
+    /**
+     * Construct new trainer with the same parameters as this trainer, but with new error threshold.
+     *
+     * @param errorThreshold New error threshold.
+     * @return new trainer with the same parameters as this trainer, but with new error threshold.
+     */
+    public MLPLocalBatchTrainer withErrorThreshold(double errorThreshold) {
+        return new MLPLocalBatchTrainer<>(loss, updaterSupplier, errorThreshold, maxIterations);
+    }
+
+    /**
+     * Construct new trainer with the same parameters as this trainer, but with new maximal iterations count.
+     *
+     * @param maxIterations New maximal iterations count.
+     * @return new trainer with the same parameters as this trainer, but with new maximal iterations count.
+     */
+    public MLPLocalBatchTrainer withMaxIterations(int maxIterations) {
+        return new MLPLocalBatchTrainer<>(loss, updaterSupplier, errorThreshold, maxIterations);
+    }
+
+    /**
+     * Set logger.
+     *
+     * @param log Logger.
+     * @return This object.
+     */
+    public MLPLocalBatchTrainer setLogger(IgniteLogger log) {
+        this.log = log;
+
+        return this;
+    }
+
+    /**
+     * Output debug message.
+     *
+     * @param msg Message.
+     */
+    private void debug(String msg) {
```

Comment:
It should be utility method. Don't use logging without disucssion with the dev-list, We should have one approach